Objects/longobject.c

```diff
@@ -207,7 +207,7 @@ _PyLong_FromNbIndexOrNbInt(PyObject *integral)
     nb = Py_TYPE(integral)->tp_as_number;
     if (nb == NULL || (nb->nb_index == NULL && nb->nb_int == NULL)) {
         PyErr_Format(PyExc_TypeError,
-                     "an integer is required (got type %.200s)",
+                     "an integer or a float is required (got type %.200s)",
```

Comment:
I know this is trivial, but it seems more consistent with existing code to say `"an integer or float is required"` (that is, drop the `"a"`). Would you mind making that change, too? I apologize for not catching that earlier.